app/javascript/components/schedule-form/index.jsx

```diff
@@ -0,0 +1,170 @@
+import { getSubmitData } from './helper';
+
+const ScheduleForm = ({
+  recordId, actionOptions, filterOptions,
+}) => {
+  const newRecord = recordId === 'new';
+
+  const [data, setData] = useState({
+    initialValues: { filter_type: 'all', action: 'vm' },
+    isLoading: true,
+    timezoneOptions: [],
+    subActionOptions: [],
+    targetOptions: [],
+    requestOptions: [],
+    objectTypeOptions: [],
+    objectItemOptions: [],
+    everyTimeOptions: [],
+    hideTarget: true,
+    hideFilterType: false,
+    hideAutomationFields: true,
+    hideObjectItem: true,
+    hideEveryTime: true,
+  });
+
+  /** Function which returns the data needed for timezone drop-down-list */
+  const timeZoneData = (timezones) => timezones.map((timezone) => ({ value: timezone.name, label: timezone.description }));
+
+  // const promise = useMemo(() => )
+
+  useEffect(() => {
+    if (newRecord) {
+      API.get('/api').then(({ timezones }) => {
+        setData({
+          ...data,
+          isLoading: false,
+          timezoneOptions: timeZoneData(timezones),
+          subActionOptions: filterOptions[0].vm_options,
+        });
+      });
+    } else {
+      Promise.all([
+        http.get(`/ops/schedule_form_fields/${recordId}`),
+        API.get('/api'),
+      ]).then(([scheduleResponse, { timezones }]) => {
+        console.log(scheduleResponse);
+        // const a = {
+        //   action_typ: scheduleResponse.action_type,
+
+        // };
+        // const reStructureData = { ...scheduleResponse, a };
+        // const formData = { ...data.initialValues, reStructureScheduleResponse(scheduleResponse) };
+        data.initialValues.action_typ = scheduleResponse.action_type;
+        data.initialValues.filter_type = scheduleResponse.filter_type;
+        data.initialValues.target_type = scheduleResponse.filter_value;
+        data.initialValues.instance_name = scheduleResponse.instance_name;
+        data.initialValues.object_message = scheduleResponse.object_message;
+        data.initialValues.object_request = scheduleResponse.object_request;
+        data.initialValues.name = scheduleResponse.schedule_name;
+        data.initialValues.description = scheduleResponse.schedule_description;
+        data.initialValues.enabled = scheduleResponse.schedule_enabled;
+        data.initialValues.start_date = scheduleResponse.schedule_start_date;
+        data.initialValues.start_hour = `${scheduleResponse.schedule_start_hour}:${scheduleResponse.schedule_start_min}`;
+        data.initialValues.timezone = scheduleResponse.schedule_time_zone;
+        data.initialValues.timer_type = scheduleResponse.schedule_timer_type;
+        data.initialValues.timer_value = scheduleResponse.schedule_timer_value;
+        data.initialValues.target_class = scheduleResponse.target_class;
+        data.initialValues.target_id = scheduleResponse.target_id;
+        data.initialValues.zone_id = scheduleResponse.zone_id;
```

Comment:
Here couldnt we do something like
```
data.initialValues = {
... (scheduleResponse)
}
```

Reply:
yes. we had moved the logic to `helper.js` file and changed the logic.. another problem here is that, most of keys we get from the `scheduleResponse` are different from `initialValues`. eg:

`data.initialValues.action_typ = scheduleResponse.action_type;`